### PR TITLE
feat(ui-test): cover email

### PR DESCRIPTION
**Pull request by arein**
Covers full end-2-end using Mailsac to get the actual one-time-password and device registration flow.
Covers only ethers for now. Wagmi is blocked (see code comments)

# Breaking Changes

N/A

# Changes

- feat(ui-test): cover email
- fix:
- chore:

# Associated Issues

closes #...

### CODE REVIEW

File: apps/laboratory/tests/shared/utils/email.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  async getNewMessage(email: string) {
+    const timeout = new Promise((_, reject) => {
+      setTimeout(() => {
+        reject('timeout')
+      }, 15000)
+    })
+
+    const messagePoll = new Promise(resolve => {
+      const interval = setInterval(async () => {
+        const messages = await this.mailsac.messages.listMessages(email)
```
Comment (chris13524) [inline apps/laboratory/tests/shared/utils/email.ts:25]: Nice find with this tool!

Looks like we have less than 500 test runs per month in the free plan so it's possible we might have to pay at some point: https://docs.mailsac.com/en/latest/services/api_calls/api_calls.html

**Review by chris13524 — APPROVED** [review]

File: apps/laboratory/tests/shared/utils/email.ts (hunk 1 above)
Comment (arein) [reply apps/laboratory/tests/shared/utils/email.ts:25]: Just signed up for the paid plan after I didn't see any initial pushback.
Paid plan gives us 25k interactions and 50 reserved emails.

I'm using 10 of these 50 for a naive collission avoidance strategy where each test cleans up the email's mailbox and then runs. Could be more sophisticated about this but works for now.

File: apps/laboratory/tests/email.spec.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  await modalPage.enterOTP(otp)
+  await modalValidator.expectConnected()
+})
+
+testMEmail('it should sign', async ({ modalPage, modalValidator }) => {
+  testMEmail.skip(modalPage.library === 'wagmi', 'Tests are flaky on wagmi')
+  await modalPage.sign()
+  await modalPage.appoveSign()
```
Comment (svenvoskamp) [inline apps/laboratory/tests/email.spec.ts:55]: small typo `appoveSign` -> `approveSign`

**Review by svenvoskamp — APPROVED** [review]
LGTM, but please fix typos above ^

File: apps/laboratory/tests/shared/pages/ModalPage.ts [hunk 3]
```diff
@@ -42,6 +72,17 @@ export class ModalPage {
     await this.page.getByTestId('sign-message-button').click()
   }
 
+  async appoveSign() {
```
Comment (svenvoskamp) [inline apps/laboratory/tests/shared/pages/ModalPage.ts:75]: small typo `appoveSign` -> `approveSign`

**Pull request merged**